Charon/filetypes/CuraPackage.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# libCharon is released under the terms of the LGPLv3 or higher.
+import io
+import re
+from collections import OrderedDict
+from typing import List, Set
+from zipfile import ZipFile
+
+from Charon.OpenMode import OpenMode
+from Charon.filetypes.OpenPackagingConvention import OpenPackagingConvention
+
+
+##  A container file that contains Cura resources that belong together.
+#   Cura can install these packages to extend the program with user generated content.
+#   Examples are Cura plugins or a set of material and quality profiles.
+class CuraPackage(OpenPackagingConvention):
+
+    # The following files should be ignored when adding a plugin directory.
+    PLUGIN_IGNORED_FILES = {r"__pycache__", r"\.qmlc", r"\.pyc"}
+    REQUIRED_METADATA_FIELDS = {"package_id", "author/author_id"}
+
+    _global_metadata_file = "/package.json"
+    _metadata_relationship_type = "http://schemas.ultimaker.org/package/2018/relationships/curapackage_metadata"
+    mime_type = "application/x-curapackage"
+
+    # File aliases for quick and easy access.
+    _aliases = OrderedDict([
+        (r"/materials", "/files/resources/materials"),
+        (r"/qualities", "/files/resources/qualities"),
+        (r"/definitions", "/files/resources/definitions"),
+        (r"/plugins", "/files/plugins")
+    ])
+
+    ##  Gets a list of paths to material files in the package.
+    def getMaterials(self) -> List[str]:
+        return self.listPaths("/materials")
+
+    ##  Add a new material file.
+    #   \param material_data The data of the material file in bytes.
+    #   \param package_filename The location to write the data to inside the package.
```

Comment:
Shouldn't it just be "filename" (or even; name)  and this convenience function then ensures that all the right things happen (eg; it is put in the correct subfolder, right extensions, etc)?

Reply:
That would be better indeed. It's actually the case but not param name is not clear about it.